src/core/providers/ogr/qgsgeopackageproviderconnection.cpp

```diff
@@ -432,10 +432,15 @@ QList<QVariantList> QgsGeoPackageProviderConnection::executeGdalSqlPrivate( cons
 QList<QgsVectorDataProvider::NativeType> QgsGeoPackageProviderConnection::nativeTypes() const
 {
   QList<QgsVectorDataProvider::NativeType> types;
-  QgsVectorLayer vl { uri(), QStringLiteral( "temp_layer" ), QStringLiteral( "ogr" ) };
+  QgsVectorLayer::LayerOptions options { false, true };
+  options.skipCrsValidation = true;
+  const QgsVectorLayer vl { uri(), QStringLiteral( "temp_layer" ), QStringLiteral( "ogr" ), options };
   if ( ! vl.isValid() || ! vl.dataProvider() )
   {
-    throw QgsProviderConnectionException( QObject::tr( "Error retrieving native types for %1: %2" ).arg( uri() ).arg( vl.dataProvider()->errors().join( '\n' ) ) );
+    const QString errorCause { vl.dataProvider() &&vl.dataProvider()->hasErrors() ?
```

Comment:
astyle seems to suggest removing the space after `&&` operator when inside a curly braces initialization. It's not that common a case, but it looks kind of ugly. Could this be mitigated in astyle?

Reply:
Probably easiest just to avoid the brace 😉

Reply:
Yeah, we have these nice tools that like to live their own lives and tell us what it's right and wrong. 

I think it's a kind of a human predominance thing that I insist to use `{}`.